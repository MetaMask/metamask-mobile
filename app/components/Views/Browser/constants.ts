import { Dimensions } from 'react-native';
import Device from '../../../util/device';

const margin = 16;

export const THUMB_WIDTH = Dimensions.get('window').width / 2 - margin * 2;
export const THUMB_HEIGHT = Device.isIos()
  ? THUMB_WIDTH * 1.81
  : THUMB_WIDTH * 1.48;
