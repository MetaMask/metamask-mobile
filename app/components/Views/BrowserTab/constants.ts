// eslint-disable-next-line import/prefer-default-export
export const IPFS_GATEWAY_DISABLED_ERROR =
  'IPFS gateway is disabled on security and privacy settings';
