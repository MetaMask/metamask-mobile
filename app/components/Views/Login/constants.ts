export const PASSCODE_NOT_SET_ERROR = 'Error: Passcode not set.';
export const WRONG_PASSWORD_ERROR = 'Error: Decrypt failed';
export const WRONG_PASSWORD_ERROR_ANDROID =
  'Error: error:1e000065:Cipher functions:OPENSSL_internal:BAD_DECRYPT';
export const VAULT_ERROR = 'Cannot unlock without a previous vault.';
export const DENY_PIN_ERROR_ANDROID = 'Error: Error: Cancel';
export const JSON_PARSE_ERROR_UNEXPECTED_TOKEN = 'Error: JSON Parse error';
