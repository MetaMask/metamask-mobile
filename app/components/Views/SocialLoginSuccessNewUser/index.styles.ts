import { StyleSheet } from 'react-native';
import { Theme } from '../../../util/theme/models';
import Device from '../../../util/device';

const createStyles = (colors: Theme['colors']) =>
  StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: colors.background.default,
      paddingHorizontal: 24,
      paddingTop: 60,
      paddingBottom: 40,
    },
    content: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
    },
    animationContainer: {
      width: Device.isMediumDevice() ? 200 : 240,
      height: Device.isMediumDevice() ? 200 : 240,
      marginBottom: 40,
      alignItems: 'center',
      justifyContent: 'center',
    },
    foxAnimation: {
      width: '100%',
      height: '100%',
    },
    textContainer: {
      alignItems: 'center',
      paddingHorizontal: 20,
    },
    title: {
      textAlign: 'center',
      marginBottom: 16,
      color: colors.text.default,
    },
    buttonContainer: {
      paddingTop: 20,
    },
  });

export default createStyles;
