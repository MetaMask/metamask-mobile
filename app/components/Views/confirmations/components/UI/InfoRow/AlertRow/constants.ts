export enum RowAlertKey {
  Blockaid = 'blockaid',
}
