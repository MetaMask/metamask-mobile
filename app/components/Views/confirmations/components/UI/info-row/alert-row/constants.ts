export enum RowAlertKey {
  Blockaid = 'blockaid',
  EstimatedFee = 'estimatedFee',
  RequestFrom = 'requestFrom',
  AccountTypeUpgrade = 'accountTypeUpgrade',
}
