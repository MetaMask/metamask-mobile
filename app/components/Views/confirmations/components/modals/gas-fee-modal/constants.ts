export const MODALS = {
  ESTIMATES: 'estimatesModal',
  ADVANCED_EIP1559: 'advancedEIP1559Modal',
  ADVANCED_GAS_PRICE: 'advancedGasPriceModal',
};
