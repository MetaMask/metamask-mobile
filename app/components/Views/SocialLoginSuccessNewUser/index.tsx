import React, { useLayoutEffect } from 'react';
import { View, ScrollView } from 'react-native';
import {
  useNavigation,
  useRoute,
  ParamListBase,
} from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import LottieView, { AnimationObject } from 'lottie-react-native';

import Button, {
  ButtonSize,
  ButtonVariants,
  ButtonWidthTypes,
} from '../../../component-library/components/Buttons/Button';
import Text from '../../../component-library/components/Texts/Text';
import { TextVariant } from '../../../component-library/components/Texts/Text/Text.types';
import { useTheme } from '../../../util/theme';
import { getTransparentOnboardingNavbarOptions } from '../../UI/Navbar';
import { strings } from '../../../../locales/i18n';
import Routes from '../../../constants/navigation/Routes';
import { PREVIOUS_SCREEN, ONBOARDING } from '../../../constants/navigation';

import CelebratingFox from '../../../animations/Celebrating_Fox.json';
import createStyles from './index.styles';

const SocialLoginSuccessNewUser: React.FC = () => {
  const navigation = useNavigation<StackNavigationProp<ParamListBase>>();
  const route = useRoute();
  const { colors } = useTheme();
  const styles = createStyles(colors);

  const { accountName, oauthLoginSuccess, onboardingTraceCtx } =
    (route.params as {
      accountName?: string;
      oauthLoginSuccess?: boolean;
      onboardingTraceCtx?: unknown;
    }) || {};

  useLayoutEffect(() => {
    navigation.setOptions(
      getTransparentOnboardingNavbarOptions(colors, undefined, false),
    );
  }, [navigation, colors]);

  const handleSetMetaMaskPin = () => {
    navigation.replace(Routes.ONBOARDING.CHOOSE_PASSWORD, {
      [PREVIOUS_SCREEN]: ONBOARDING,
      oauthLoginSuccess,
      onboardingTraceCtx,
      accountName,
    });
  };

  return (
    <ScrollView contentContainerStyle={styles.container}>
      <View style={styles.content}>
        <View style={styles.animationContainer}>
          <LottieView
            style={styles.foxAnimation}
            autoPlay
            loop
            source={CelebratingFox as AnimationObject}
            resizeMode="contain"
          />
        </View>

        <View style={styles.textContainer}>
          <Text variant={TextVariant.DisplayMD} style={styles.title}>
            {strings('social_login_success.title')}
          </Text>
        </View>
      </View>

      <View style={styles.buttonContainer}>
        <Button
          variant={ButtonVariants.Primary}
          size={ButtonSize.Lg}
          width={ButtonWidthTypes.Full}
          onPress={handleSetMetaMaskPin}
          label={strings('social_login_success.set_metamask_pin')}
        />
      </View>
    </ScrollView>
  );
};

export default SocialLoginSuccessNewUser;
