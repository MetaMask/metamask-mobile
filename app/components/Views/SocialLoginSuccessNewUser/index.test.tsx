import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { useNavigation, useRoute } from '@react-navigation/native';
import SocialLoginSuccessNewUser from './index';
import Routes from '../../../constants/navigation/Routes';
import { PREVIOUS_SCREEN, ONBOARDING } from '../../../constants/navigation';

// Mock dependencies
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
  useRoute: jest.fn(),
}));

const mockUseTheme = jest.fn();
jest.mock('../../../util/theme', () => ({
  useTheme: mockUseTheme,
}));

jest.mock('../../../../locales/i18n', () => ({
  strings: jest.fn((key) => key),
}));

jest.mock('../../UI/Navbar', () => ({
  getTransparentOnboardingNavbarOptions: jest.fn(() => ({})),
}));

jest.mock('lottie-react-native', () => 'LottieView');

jest.mock('../../../animations/Celebrating_Fox.json', () => ({}));

const mockNavigation = {
  replace: jest.fn(),
  setOptions: jest.fn(),
};

const mockRoute = {
  params: {
    accountName: 'test@example.com',
    oauthLoginSuccess: true,
    onboardingTraceCtx: { traceId: 'test-trace' },
  },
};

const mockTheme = {
  colors: {
    background: { default: '#FFFFFF' },
    text: { default: '#000000' },
  },
};

describe('SocialLoginSuccessNewUser', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    (useNavigation as jest.Mock).mockReturnValue(mockNavigation);
    (useRoute as jest.Mock).mockReturnValue(mockRoute);
    mockUseTheme.mockReturnValue(mockTheme);
  });

  it('should render correctly', () => {
    const { toJSON } = render(<SocialLoginSuccessNewUser />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should display the success title', () => {
    const { getByText } = render(<SocialLoginSuccessNewUser />);
    expect(getByText('social_login_success.title')).toBeTruthy();
  });

  it('should display the set pin button', () => {
    const { getByText } = render(<SocialLoginSuccessNewUser />);
    expect(getByText('social_login_success.set_metamask_pin')).toBeTruthy();
  });

  it('should render fox animation', () => {
    const { toJSON } = render(<SocialLoginSuccessNewUser />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should call navigation.replace when Set MetaMask pin button is pressed', () => {
    const { getByText } = render(<SocialLoginSuccessNewUser />);
    const button = getByText('social_login_success.set_metamask_pin');

    fireEvent.press(button);

    expect(mockNavigation.replace).toHaveBeenCalledWith(
      Routes.ONBOARDING.CHOOSE_PASSWORD,
      {
        [PREVIOUS_SCREEN]: ONBOARDING,
        oauthLoginSuccess: true,
        onboardingTraceCtx: { traceId: 'test-trace' },
        accountName: 'test@example.com',
      },
    );
  });

  it('should handle missing route params gracefully', () => {
    (useRoute as jest.Mock).mockReturnValue({ params: undefined });

    const { getByText } = render(<SocialLoginSuccessNewUser />);
    const button = getByText('social_login_success.set_metamask_pin');

    fireEvent.press(button);

    expect(mockNavigation.replace).toHaveBeenCalledWith(
      Routes.ONBOARDING.CHOOSE_PASSWORD,
      {
        [PREVIOUS_SCREEN]: ONBOARDING,
        oauthLoginSuccess: undefined,
        onboardingTraceCtx: undefined,
        accountName: undefined,
      },
    );
  });

  it('should handle partial route params', () => {
    (useRoute as jest.Mock).mockReturnValue({
      params: {
        accountName: 'partial@test.com',
      },
    });

    const { getByText } = render(<SocialLoginSuccessNewUser />);
    const button = getByText('social_login_success.set_metamask_pin');

    fireEvent.press(button);

    expect(mockNavigation.replace).toHaveBeenCalledWith(
      Routes.ONBOARDING.CHOOSE_PASSWORD,
      {
        [PREVIOUS_SCREEN]: ONBOARDING,
        oauthLoginSuccess: undefined,
        onboardingTraceCtx: undefined,
        accountName: 'partial@test.com',
      },
    );
  });

  it('should set navigation options on mount', () => {
    render(<SocialLoginSuccessNewUser />);
    expect(mockNavigation.setOptions).toHaveBeenCalledWith({});
  });

  it('should use theme colors for styling', () => {
    const customTheme = {
      colors: {
        background: { default: '#123456' },
        text: { default: '#ABCDEF' },
      },
    };
    mockUseTheme.mockReturnValue(customTheme);

    const { toJSON } = render(<SocialLoginSuccessNewUser />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render without crashing when all props are provided', () => {
    expect(() => render(<SocialLoginSuccessNewUser />)).not.toThrow();
  });

  it('should render without crashing when no props are provided', () => {
    (useRoute as jest.Mock).mockReturnValue({});
    expect(() => render(<SocialLoginSuccessNewUser />)).not.toThrow();
  });
});
