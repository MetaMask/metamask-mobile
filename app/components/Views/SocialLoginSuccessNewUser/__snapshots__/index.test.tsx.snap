// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLoginSuccessNewUser renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "paddingBottom": 40,
      "paddingHorizontal": 24,
      "paddingTop": 60,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 240,
            "justifyContent": "center",
            "marginBottom": 40,
            "width": 240,
          }
        }
      >
        <LottieView
          autoPlay={true}
          loop={true}
          resizeMode="contain"
          source={{}}
          style={
            {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#121314",
              "fontFamily": "Geist Bold",
              "fontSize": 32,
              "letterSpacing": 0,
              "lineHeight": 40,
              "marginBottom": 16,
              "textAlign": "center",
            }
          }
        >
          social_login_success.title
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "paddingTop": 20,
        }
      }
    >
      <TouchableOpacity
        accessibilityRole="button"
        accessible={true}
        activeOpacity={1}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#121314",
            "borderRadius": 12,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingHorizontal": 16,
          }
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#ffffff",
              "fontFamily": "Geist Medium",
              "fontSize": 16,
              "letterSpacing": 0,
              "lineHeight": 24,
            }
          }
        >
          social_login_success.set_metamask_pin
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</RCTScrollView>
`;

exports[`SocialLoginSuccessNewUser renders fox animation 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "paddingBottom": 40,
      "paddingHorizontal": 24,
      "paddingTop": 60,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 240,
            "justifyContent": "center",
            "marginBottom": 40,
            "width": 240,
          }
        }
      >
        <LottieView
          autoPlay={true}
          loop={true}
          resizeMode="contain"
          source={{}}
          style={
            {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#121314",
              "fontFamily": "Geist Bold",
              "fontSize": 32,
              "letterSpacing": 0,
              "lineHeight": 40,
              "marginBottom": 16,
              "textAlign": "center",
            }
          }
        >
          social_login_success.title
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "paddingTop": 20,
        }
      }
    >
      <TouchableOpacity
        accessibilityRole="button"
        accessible={true}
        activeOpacity={1}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#121314",
            "borderRadius": 12,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingHorizontal": 16,
          }
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#ffffff",
              "fontFamily": "Geist Medium",
              "fontSize": 16,
              "letterSpacing": 0,
              "lineHeight": 24,
            }
          }
        >
          social_login_success.set_metamask_pin
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</RCTScrollView>
`;

exports[`SocialLoginSuccessNewUser uses theme colors for styling 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "paddingBottom": 40,
      "paddingHorizontal": 24,
      "paddingTop": 60,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 240,
            "justifyContent": "center",
            "marginBottom": 40,
            "width": 240,
          }
        }
      >
        <LottieView
          autoPlay={true}
          loop={true}
          resizeMode="contain"
          source={{}}
          style={
            {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#121314",
              "fontFamily": "Geist Bold",
              "fontSize": 32,
              "letterSpacing": 0,
              "lineHeight": 40,
              "marginBottom": 16,
              "textAlign": "center",
            }
          }
        >
          social_login_success.title
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "paddingTop": 20,
        }
      }
    >
      <TouchableOpacity
        accessibilityRole="button"
        accessible={true}
        activeOpacity={1}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#121314",
            "borderRadius": 12,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingHorizontal": 16,
          }
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#ffffff",
              "fontFamily": "Geist Medium",
              "fontSize": 16,
              "letterSpacing": 0,
              "lineHeight": 24,
            }
          }
        >
          social_login_success.set_metamask_pin
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</RCTScrollView>
`;
